src/linkedList/LinkedList.java

```diff
@@ -66,162 +143,223 @@ public int getLength()
-    * 
+    *
     * @param someNode The ListNode you want to add to the LinkedList
-    * @return The Integer length of the list after adding the ListNode. 0 if 
-    * the list is full. -1 if the ListNode.next is not empty.
+    * @return The Integer length of the list after adding the ListNode. 0 if the
+    * list is full. -1 if the ListNode.next is not empty.
     */
-   public int addEntryToTail(ListNode someNode)
+   public int addEntryToTail(E someData)
    {
-      if(someNode.getNodeNext() != null)
-      {
-         return -1;
-      }
-      if(this.length < this.maxSize)
+
+      if (this.length < this.maxSize)
       {
-         if(this.tail == null)
+         ListNode someNode = new ListNode<>(someData, null);
+         if (this.tail == null)
          {
             /* we have an empty list */
             this.head = someNode;
             this.tail = someNode;
-            this.head = someNode;
-            this.length++;
-            return this.length;
-         }
-         else if(this.tail.setNodeNext(someNode))
+         } else
          {
+            this.tail.next = someNode;
             this.tail = someNode;
-            this.length++;
-            return this.length;
-         }
-         else
-         {
-            return -1;
          }
+         this.length++;
+         return this.length;
+      } else
+      {
+         return 0;
+      }
+   }
+
+   /**
+    * Accessor function to return the Object being stored at the head of the 
+    * list. The Object remains in the list.
+    * @return The Object at the head of the list - null if the list is empty.
+    */
+   public E getFirst()
+   {
+      if (this.isEmpty())
+      {
+         return null;
+      }
+      return this.head.getNodeData();
+   }
+   
+   /**
+    * Accessor function to return the Object being stored at the tail of the 
+    * list. The Object remains in the list.
+    * @return The Object at the tail of the list - null if the list is empty.
+    */
+   public E getLast()
+   {
+      if (this.isEmpty())
+      {
+         return null;
+      }
+      return this.tail.getNodeData();
+   }
+   
+   /**
+    * Modifier function to remove a target entry from the list. This function
+    * requires that the caller pass a target object that is comparable to 
+    * another instance of that object being stored in the list.
+    * @param someTarget A comparable Object that you expect to exist in the 
+    * list.
+    * @return The Object that has been removed from the list - null if the list
+    * is empty or if the target was unmatched.
+    */
+   public E removeEntry(E someTarget)
+   {
+      ListNode prevNode;
+      E foundTarget = null;
+
+      if (this.isEmpty())
+      {
+         return null;
+      }
+
+      /* check the head first as findTargetPrev will return a false positive 
+       * in that case.
+       */
+      if (someTarget.equals(this.getFirst()))
```

Comment:
I don't think we need this if/else at all. just modify that findTargetPrev() method as mentioned below.